Core/Base/SQLAlchemyDB.py

```diff
@@ -0,0 +1,303 @@
+""" SQLAlchemyDB:
+    This module provides the BaseRSSDB class for providing standard DB interactions.
+
+    Uses sqlalchemy
+"""
+
+__RCSID__ = "$Id$"
+
+import datetime
+from sqlalchemy import create_engine, desc, exc
+from sqlalchemy.engine.reflection import Inspector
+from sqlalchemy.orm import sessionmaker
+from sqlalchemy.orm.query import Query
+
+from DIRAC import gConfig, gLogger, S_OK, S_ERROR
+from DIRAC.ConfigurationSystem.Client.Utilities import getDBParameters
+
+
+class SQLAlchemyDB(object):
+  """
+    Base class that defines some of the basic DB interactions.
+  """
+
+  def __init__(self, logSubName):
+    """c'tor
+
+    :param self: self reference
+    """
+
+    self.log = gLogger.getSubLogger(logSubName)
```

Comment:
you can do `self.log = gLogger.getSubLogger(__name__)` and remove `logSubName`

Reply:
This would cause errors. Because `__name__` will have the value `SQLAlchemyDB.py`.

I would have to do `self.log = gLogger.getSubLogger(self.__class__)` instead.

Reply:
The advantage of using `__name__` is that it contains the full name including the path for the module (`DIRAC.Core.Utilities.SQLAlchemyDB`), which let's one do hierarchical selection of log output, e.g., silence everything from Utilities, or the opposite. Thus each module should have their own `__name__`'ed logging instance

Reply:
@andresailer In that cast `self.log = gLogger.getSubLogger(self.__class__.__module__)` is the way rather thatn `__name__`